File: backend/internal/models/comment.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Comment struct {
	ID        uint           `gorm:"primarykey" json:"id"`
	PostID    uint           `gorm:"not null;index" json:"post_id"`
	UserID    uint           `gorm:"not null;index" json:"user_id"`
	Content   string         `gorm:"type:text;not null" json:"content" validate:"required,max=280"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`

	// リレーション
	Post Post `gorm:"foreignKey:PostID" json:"post,omitempty"`
	User User `gorm:"foreignKey:UserID" json:"user,omitempty"`
}

File: frontend/src/types/comment.ts
import type { User } from './user';

// コメント型定義
export interface Comment {
  id: number;
  user: User;
  post_id: number;
  content: string;
  created_at: string;
  updated_at?: string;
}

// コメント作成リクエスト型
export interface CreateCommentRequest {
  content: string;
}
